cpp/src/reductions/nth_element.cu

```diff
@@ -0,0 +1,138 @@
+           dcol] __device__(auto n) {
+            *dres = static_cast<OutputType>(dcol.element<ElementType>(n));
+            *dvalid = dcol.is_valid(n);
+          });
+    }
+    return std::unique_ptr<scalar>(result);
+  }
+
+  template <typename OutputType>
+  static constexpr bool is_supported_s() {
+    return std::is_same<OutputType, cudf::string_view>::value &&
+           std::is_same<ElementType, cudf::string_view>::value;
+  }
+
+  template <typename OutputType>
+  static constexpr bool is_not_supported() {
+    return  !is_supported_v<OutputType>() ||
+           (!is_fixed_width<OutputType>() && !is_supported_s<OutputType>());
+  }
+
+  template <typename OutputType>
+  std::enable_if_t<is_supported_s<OutputType>(), std::unique_ptr<scalar>>
+  operator()(column_device_view const& dcol,
+             size_type n,
+             rmm::mr::device_memory_resource* mr, cudaStream_t stream) {
+    bool const n_valid = n < dcol.size() && n >= -dcol.size();
+    using OutputPair = thrust::pair<OutputType, bool>;
+    OutputPair host_result{{}, n_valid};
+    if (n_valid) {
+      n = (n < 0 ? dcol.size() + n : n);
+      if (dcol.nullable()) {
+        auto it = dcol.pair_begin<ElementType, true>() + n;
+        rmm::device_scalar<OutputPair> dev(host_result, stream, mr);
+        thrust::copy(rmm::exec_policy(stream)->on(stream), it, it + 1, dev.data());
+        host_result = dev.value();
+      } else {
+        auto it = dcol.pair_begin<ElementType, false>() + n;
+        rmm::device_scalar<OutputPair> dev(host_result, stream, mr);
+        thrust::copy(rmm::exec_policy(stream)->on(stream), it, it + 1, dev.data());
+        host_result = dev.value();
+      }
+    }
+    using ScalarType = cudf::experimental::scalar_type_t<OutputType>;
+    // only for string_view, data is copied
+    auto result = new ScalarType(host_result.first, host_result.second, stream, mr);
+    return std::unique_ptr<scalar>(result);
+  }
+
+  template <typename OutputType>
+  std::enable_if_t<is_not_supported<OutputType>(), std::unique_ptr<scalar>>
+  operator()(column_device_view const& dcol,
+             size_type n,
+             rmm::mr::device_memory_resource* mr,
+             cudaStream_t stream) {
+    CUDF_FAIL("Only fixed type is supported for nth element");
+  }
+};
+
+struct element_dispatch_nth_element_functor {
+  template <typename ElementType>
+  std::unique_ptr<cudf::scalar> operator()(column_device_view const& dcol,
+                                           data_type const output_dtype,
+                                           size_type n,
+                                           rmm::mr::device_memory_resource* mr,
+                                           cudaStream_t stream) {
+    return cudf::experimental::type_dispatcher(
+        output_dtype, nth_element_functor<ElementType>{}, 
+        dcol, n, mr, stream);
+  }
+};
+}  // namespace anonymous
+
+std::unique_ptr<cudf::scalar> nth_element(column_view const& col,
```

Comment:
I don't think this really makes sense as a "reduction" in C++. We really just want the [`get_element`](https://github.com/rapidsai/cudf/issues/4618) function and a Python `nth` reduction can call that. 

CC @devavret

Reply:
Now the question is, should I target it for 0.14?

Reply:
I can take it for 0.14. 
reduction `nth` has an extra feature of type conversion from ElementType to OutputType.
I will split this functionality to `get_element` and `static_cast<OutputType>` functor.

Reply:
If Pandas series doesn't have an `nth`, then why are we adding a `nth` column reduction?

Reply:
It's only  for completeness of all present aggregations in reduce.
`nth` is new code.  Rest all aggregations are mere function calls to existing code.
issue #4707 description had question if this is really needed or not. .

Reply:
Let's not add features that aren't needed.

Reply:
New features should be 0.15.